eclass/coreos-cargo.eclass

```diff
@@ -76,15 +76,36 @@ coreos-cargo_src_unpack() {
 
 	# Compile for the built-in target, using the SDK cross-tools.
 	export RUST_TARGET=$(rust_builtin_target "${CHOST}")
-	cat <<- EOF >> "${ECARGO_HOME}/config"
-
-	[build]
-	target = "${RUST_TARGET}"
-
-	[target.${RUST_TARGET}]
-	ar = "${TARGET_AR}"
-	linker = "${TARGET_CC}"
-	EOF
+	local -a config_lines
+	local build_amended=0
+	local target_rust_target_amended=0
+	local REPLY
+	readonly b_header='[build]'
+	readonly t_header="[target.${RUST_TARGET}]"
+	readonly target_line="target = \"${RUST_TARGET}\""
+	readonly ar_line="ar = \"${TARGET_AR}\""
+	readonly linker_line="linker = \"${TARGET_CC}\""
+	while read -r; do
+		config_lines+=("${REPLY}")
+		case "${REPLY}" in
+			"${b_header}")
+				config_lines+=("${target_line}")
+				build_amended=1
+				;;
+			"${t_header}")
+				config_lines+=("${ar_line}")
+				config_lines+=("${linker_line}")
+				target_rust_target_amended=1
+				;;
+		esac
+	done <"${ECARGO_HOME}/config"
+	if [[ "${build_amended}" -eq 0 ]]; then
+	    config_lines+=('' "${b_header}" "${target_line}")
+	fi
+	if [[ "" -eq 0 ]]; then
```

Comment:
what is this?

Reply:
Previously, coreos-cargo appended a `[build]` and `[target.whatever]` sections to some cargo config and that worked. But after updating cargo eclass in portage stable, the config has changed too - it has a `[build]` section in the middle of the file. So coreos-cargo appended another `[build]` section and cargo bailed out complaining about an invalid toml file.

This is a workaround to keep just one `[build]` section until I maybe have some deeper look at our way of building rust packages. I hope that this is just some nonsense now and can be dropped in favor of cargo.

Reply:
Oh, wait. You mean the `[[ "" -eq 0 ]]` part. Whoops.

Reply:
Funny - `[[ "" -eq 0 ]]` seems to be always true, so we were always appending the `[target.whatever]` section to the end of file. Which is fine, because it did not exist in config generated by cargo eclass. But will fix, it was supposed to be `[[ "${target_rust_target_amended}" -eq 0 ]]`.